### PR TITLE
How to declare simple observer with function

### LINKED ISSUE #2404: Update accepted types for simple observer
In https://github.com/Polymer/polymer/pull/4574 we updated the accepted values of `observer` for properties to also accept a function. The documentation of https://www.polymer-project.org/2.0/docs/devguide/properties should be updated to reflect this change.

### PR DESCRIPTION
**Pull request by katejeffreys**
Fixes #2404

### CODE REVIEW

**Comment by katejeffreys** [comment]
@arthurevans and/or @TimvdLippe can has sanity-check?

File: app/2.0/docs/devguide/observers.md [hunk 1]
```diff
@@ -100,15 +104,34 @@ The observer method is usually defined on the class itself, although an observer
...
+
+  static get properties() {
+    return {
+      active: {
+        type: Boolean,
+        // Observer method
+        observer: 
+          function(newValue, oldValue){
+            this.toggleClass('highlight', newValue);
+          }.bind(this)
```
Comment (TimvdLippe) [inline app/2.0/docs/devguide/observers.md]: I think a lambda reads clearer. Does the following work?

```js
observer(newValue, oldValue) {
  this.toggleClass('highlight', newValue);
}
```

Comment (katejeffreys) [reply app/2.0/docs/devguide/observers.md]: So this?

```
observer: 
  function observer(newValue, oldValue) {
    this.toggleClass('highlight', newValue);
  }.bind(this)
```

Or ...?

Comment (TimvdLippe) [reply app/2.0/docs/devguide/observers.md]: Like this:

```js
static get properties() {
  return {
    active: {
      type: Boolean,
      // Observer method
      observer(newValue, oldValue) {
          this.toggleClass('highlight', newValue);
      }
  }
}
```

A running example: http://jsbin.com/xisifaboso/edit?html,console,output This is using the "object method shorthand definition" in ES2015: https://developer.mozilla.org/nl/docs/Web/JavaScript/Reference/Functions/Method_definitions

Comment (katejeffreys) [reply app/2.0/docs/devguide/observers.md]: Cool, didn't know that was a thing XD

Comment (TimvdLippe) [reply app/2.0/docs/devguide/observers.md]: Yeah it is fairly new and neatly concise 😄  This PR now LGTM fully!

Comment (katejeffreys) [reply app/2.0/docs/devguide/observers.md]: Great, thanks!

**Pull request merged**